packages/block-editor/src/components/heading-level-control/README.md

```diff
@@ -0,0 +1,56 @@
+## Heading Level Control
+
+The `HeadingLevelControl` component renders a dropdown menu that displays heading level options for the selected block.
+
+This component renders a dropdown menu that displays heading level options for the selected block. This component is used to set the level of the heading in a block and also allows to use a `paragraph` element by setting the level to zero (`0`). The block's `edit` function would also need to handle the zero value. An example of this would be `const TagName = level === 0 ? 'p' : `h${ level }`;`
+
+See the Heading block for an example usage.
+
+## Development guidelines
+
+### Usage
+
+Renders a heading level control with options.
+
+```jsx
+import { BlockControls, HeadingLevelControl } from '@wordpress/block-editor';
+
+const MyHeadingLevelControl = ( { attributes, setAttributes } ) => (
+	<BlockControls group="block">
+		<HeadingLevelControl
+			value={ attributes.level }
+			onChange={ ( next ) => {
+				setAttributes( { level: next } );
+			} }
+		/>
+	</BlockControls>
+);
+```
+
+### Props
+
+#### `selectedLevel`
+
+-   **Type:** `Number`
+-   **Options:**: `1`, `2`, `3`, `4`, `5`, `6` and `0` if `isParagraphAllowed` is set to `true`
+
+The current selected level of the block. Only the `Options` listed above are supported.
+
+#### `onChange`
+
+-   **Type:** `Function`
+-   **Required:** Yes
+
+A callback function invoked when the toolbar's heading value is changed via an interaction with any of the toolbar's buttons. Called with the new level value as the only argument.
+
+#### `isParagraphAllowed`
```

Comment:
I feel this is not a great prop, maybe it should be "additionalOptions" or something like that?

Reply:
You mean like an options `object` like `{ allowParargraphs: true }`? I had thought about adding the allowed levels as an option but felt a bit weird due to the convention of `paragraph === level 0`.